portfolio/src/main/webapp/script.js

```diff
@@ -23,6 +23,12 @@ function addRandomGreeting() {
   const greeting = greetings[Math.floor(Math.random() * greetings.length)];
 
   // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+}
+
+async function showGreeting() {
+    const responseFromServer = await fetch('/hello');
+    const textFromResponse = await responseFromServer.text();
+
```

Comment:
Thing to keep in my mind about defensive programming to see if the value returned from server is valid. In this case make sure that it's not an empty string. This will ensure the front end code is more robust. In this toy example it might not be huge issue but as the complexity of the system increase you want to keep these things in mind.